### ISSUE TITLE
Disable instantiation of modules with private ctor

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#7472]
Sanar violation: Make sure that this accessibility update is safe here

https://sonarcloud.io/project/issues?id=org.checkstyle%3Acheckstyle&issues=AW9t2w4vYD2QG1pPXIVH&open=AW9t2w4vYD2QG1pPXIVH

https://github.com/checkstyle/checkstyle/blob/284278fde3431e33701cad1c3cc6690abcec0bb6/src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java#L355-L361

`declaredConstructor.setAccessible(true);`

it was added at https://github.com/checkstyle/checkstyle/commit/c5c3b7eed3a5c00aee3225742365f6e36ab11dbd .... 
not sure why I was agree to allow change main code in favor of IDEA inspection. 

We do not need to load private stuff and overuse reflection. All IDEA complains should be suppressed or inspection is configured to not run in `tests` as there are bunch of public stuff.

If there is some other good reason, it is better to share it in this issue and mark issue on sonar as resolved.

**Comment by rnveach** [comment]
The reason is probably in that same commit. A lot of test modules were changed to private and I believe this is the reason we have the code. Without the code, all our test modules for testing checkstyle would have to be public and I am sure some other checks would ask they be private.

Test failures with code removed:
````
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   CheckerTest.testHaltOnException:938 Error message is not expected ==> expected: <Exception was thrown while processing M:\checkstyleWorkspace\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\checker\InputChecker.java> but was: <cannot initialize module com.puppycrawl.tools.checkstyle.TreeWalker - cannot initialize module com.puppycrawl.tools.checkstyle.CheckerTest$CheckWhichThrowsError - Unable to instantiate com.puppycrawl.tools.checkstyle.CheckerTest$CheckWhichThrowsError>
[ERROR]   TreeWalkerTest.testForInvalidCheckImplementation:200 Error message is unexpected ==> expected: <true> but was: <false>
[ERROR] Errors: 
[ERROR]   CheckerTest.testCacheAndFilterWhichDoesNotImplementExternalResourceHolderInterface:776->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:140 » Checkstyle
[ERROR]   CheckerTest.testCacheOnViolationSuppression:917->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:140 » Checkstyle
[ERROR]   CheckerTest.testClearCache:633 » Checkstyle cannot initialize module com.puppy...
[ERROR]   CheckerTest.testClearLazyLoadCacheInDetailAST:897->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:140 » Checkstyle
[ERROR]   CheckerTest.testExceptionWithCache:959->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:140 » Checkstyle
[ERROR]   CheckerTest.testHaltOnExceptionOff:1213->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:140 » Checkstyle
[ERROR]   CheckerTest.testTabViolation:1238->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:140 » Checkstyle
[ERROR]   CheckerTest.testTabViolationDefault:1224->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:144 » Checkstyle
[ERROR]   PackageObjectFactoryTest.testConstructorFailure:369 NullPointer
[ERROR]   TreeWalkerTest.testCheckDestroyIsCalledInTreeWalker:434->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:137 » Checkstyle
[ERROR]   TreeWalkerTest.testCheckInitIsCalledInTreeWalker:423->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:137 » Checkstyle
[ERROR]   TreeWalkerTest.testCommentCheckDestroyIsCalledInTreeWalker:445->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:137 » Checkstyle
[ERROR]   TreeWalkerTest.testRequiredTokenIsEmptyIntArray:310->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:137 » Checkstyle
[ERROR]   AbstractCheckTest.testCheck:321->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:137 » Checkstyle
[ERROR]   AbstractFileSetCheckTest.testCheck:195->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:144 » Checkstyle
[ERROR]   FileSetCheckTest.testProcessCallsFinishBeforeCallingDestroy:55->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:144 » Checkstyle
[ERROR]   FileSetCheckTest.testTranslation:44->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:144 » Checkstyle
[ERROR]   AbstractJavadocCheckTest.testAcceptableTokensFail:260->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:137 » Checkstyle
[ERROR]   AbstractJavadocCheckTest.testAcceptableTokensPass:280->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:137 » Checkstyle
[ERROR]   AbstractJavadocCheckTest.testAntlrError:156->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:137 » Checkstyle
[ERROR]   AbstractJavadocCheckTest.testCheckReuseAfterParseErrorWithFollowingAntlrErrorInSingleFile:189->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:137 » Checkstyle
[ERROR]   AbstractJavadocCheckTest.testCheckReuseAfterParseErrorWithFollowingAntlrErrorInTwoFiles:173->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:137 » Checkstyle
[ERROR]   AbstractJavadocCheckTest.testCustomTag:124->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:137 » Checkstyle
[ERROR]   AbstractJavadocCheckTest.testJavadocTagsWithoutArgs:107->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:137 » Checkstyle
[ERROR]   AbstractJavadocCheckTest.testNoWsBeforeDescriptionInJavadocTags:344->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:137 » Checkstyle
[ERROR]   AbstractJavadocCheckTest.testNumberFormatException:117->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:137 » Checkstyle
[ERROR]   AbstractJavadocCheckTest.testParsingErrors:134->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:137 » Checkstyle
[ERROR]   AbstractJavadocCheckTest.testPosition:199->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:137 » Checkstyle
[ERROR]   AbstractJavadocCheckTest.testPositionOnlyComments:220->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:137 » Checkstyle
[ERROR]   AbstractJavadocCheckTest.testPositionWithSinglelineComments:209->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:137 » Checkstyle
[ERROR]   AbstractJavadocCheckTest.testRequiredTokenIsNotInDefaultTokens:292->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:137 » Checkstyle
[ERROR]   AbstractJavadocCheckTest.testVisitLeaveToken:310->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:137 » Checkstyle
[ERROR]   AbstractJavadocCheckTest.testWrongSingletonTagInJavadoc:358->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:137 » Checkstyle
[ERROR]   AllBlockCommentsTest.testAllBlockComments:58->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:137 » Checkstyle
[ERROR]   AllSinglelineCommentsTest.testAllSinglelineComments:57->AbstractModuleTestSupport.verify:220->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:137 » Checkstyle
[ERROR]   XdocsJavaDocsTest.setUp:80->AbstractModuleTestSupport.createChecker:118->AbstractModuleTestSupport.createChecker:137 » Checkstyle
````

**Comment by romani** [comment]
Yes, but why not to keep them public as there were, and just force inspection to not raise violation on them? We can disable whole inspection on test code only.

**Comment by romani** [comment]
@rnveach please approve issue.
I think to put "bug" label on it and rename to `Disable instantiation of modules with private ctor` to make it user friendly in relese notes.

**Comment by pbludov** [comment]
fix is merged.

**Comment by romani** [comment]
@pbludov , please do not forget to put label on issue after meet of fix. Absence of label fails master build https://travis-ci.org/checkstyle/checkstyle/jobs/638774526#L557
